src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -138,15 +145,25 @@ public static bool IsPathRooted(string path)
         // only contains whitespace characters an ArgumentException gets thrown.
         public static string GetPathRoot(string path)
         {
-            if (path == null) return null;
+            if (path == null)
+                return null;
+
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
+            return new string(GetPathRoot(path.AsReadOnlySpan()));
```

Comment:
You can optimize the new string here as well, but you need to check that you got the same string back from normalize separators. Use object.ReferenceEquals and check lengths if true.

Reply:
Why do you need to check lengths if the objects are reference equals?

Reply:
```
ReadOnlySpan<char> result = GetPathRoot(path.AsReadOnlySpan());

 if (path.Length == result.Length && object.ReferenceEquals(result.DangerousGetPinnableReference(), path))
         return path;

return new string(result);
```